kasa/discover.py

```diff
@@ -1,4 +1,81 @@
-"""Discovery module for TP-Link Smart Home devices."""
+"""Discover TPLink Smart Home devices.
+
+The main entry point for this library is :func:`Discover.discover()`,
+which returns a dictionary of the found devices. The key is the IP address
+of the device and the value contains ready-to-use, SmartDevice-derived
+device object.
+
+:func:`discover_single()` can be used to initialize a single device given its
+IP address. If the :class:`DeviceConfig` of the device is already known,
+you can initialize the corresponding device class directly without discovery.
+
+The protocol uses UDP broadcast datagrams on port 9999 and 20002 for discovery.
+Legacy devices support discovery on port 9999 and newer devices on 20002.
+
+Newer devices that respond on port 20002 will most likely require TP-Link cloud
+credentials to be passed if queries or updates are to be performed on the returned
+devices.
+
+Discovery returns a dict of {ip: discovered devices}:
+
+>>> import asyncio
+>>> from kasa import Discover, Credentials
+>>>
+>>> found_devices = await Discover.discover()
+>>> [dev.model for dev in found_devices.values()]
+['KP303(UK)', 'HS110(EU)', 'L530E', 'KL430(US)', 'HS220(US)']
+
+Discovery can also be targeted to a specific broadcast address instead of
+the default 255.255.255.255:
+
+>>> found_devices = await Discover.discover(target="127.0.0.255")
+>>> print(len(found_devices))
+5
+
+Basic information is available on the device from the discovery broadcast response
+but it is important to call device.update() after discovery if you want to access
+all the attributes without getting errors or None.
+
+>>> dev = found_devices["127.0.0.3"]
+>>> dev.alias
+None
+>>> await dev.update()
+>>> dev.alias
+'Living Room Bulb'
+
+It is also possible to pass a coroutine to be executed for each found device:
+
+>>> async def print_alias(dev):
+>>>     await dev.update()
+>>>     print(f"Discovered {dev.alias}")
```

Comment:
```suggestion
>>>     print(f"Discovered {dev.alias} (model: {dev.model})")
```
maybe to add something like this (or the state?) to make more interesting?

Reply:
Done